scrapy/http/request/form.py

```diff
@@ -54,10 +55,15 @@ def _urlencode(seq, enc):
     return urlencode(values, doseq=1)
 
 
+def _create_parser_from_response(response, parser_cls):
```

Comment:
Sorry, I feel dumb - could you please explain what is `extended-html` and why is it different from `html`?

Promoting `_root` to `root` sounds good, I had to use `_root` more than once.